### PR TITLE
Put a reference to the range_class in mrb_state

### PR DESCRIPTION
**Pull request by bouk**
Can cause segfaults otherwise

### CODE REVIEW

File: include/mruby.h [hunk 1]
```diff
@@ -191,6 +191,7 @@ typedef struct mrb_state {
   struct RClass *proc_class;
   struct RClass *string_class;
   struct RClass *array_class;
+  struct RClass *range_class;
```
Comment (bouk) [inline include/mruby.h]: @matz does this need to be added to root_scan_phase ?

**Comment by matz** [comment]
I don't want to increase the size of mrb_state if possible. Could you tell us how we can cause SEGV?

**Comment by bouk** [comment]
If you look at the test, you can see that you can do `Range = Array` which will create a `TT_RANGE` with Array class. We could also save the range class as a global variable

**Comment by matz** [comment]
Sorry, I found the error reproducing code in the PR.  I think I found the work around.

**Pull request closed without merging**